alfalfa_web/server/api-v2.js

```diff
@@ -164,362 +252,402 @@ router.get("/sites/:siteId", async (req, res) => {
- *               $ref: '#/components/schemas/Error'
+ *               type: array
+ *               items:
+ *                 $ref: '#/components/schemas/pointMetadata'
  */
-router.get("/sites/:siteId/points", (req, res) => {
-  const { siteId } = req.params;
-
-  const error = validate(
-    { siteId },
-    {
-      siteId: "required|uuid"
-    }
-  );
-  if (error) return res.status(400).json({ error });
+router.post("/runs/:runId/points", async (req, res, next) => {
+  const { points, pointTypes } = req.body;
+
+  var pointsArr;
+  if (points) {
+    pointsArr = api.getPointsById(req.run, points);
+  } else if (pointTypes) {
+    pointsArr = api.getPointsByType(req.run, pointTypes);
+  }
 
-  api
-    .listPoints(siteId, undefined, true)
-    .then((data) => {
-      if (data) {
-        return res.json({ data });
-      } else {
-        return res.status(404).json({
-          error: `Site ID '${siteId}' does not exist`
-        });
-      }
+  pointsArr
+    .then((points) => {
+      return points.map(api.formatPoint);
     })
-    .catch(() => res.sendStatus(500));
+    .then((payload) => {
+      res.status(200).json({ payload });
+    })
+    .catch(next);
 });
 
 /**
  * @openapi
- * /sites/{siteId}/points/inputs:
+ * /runs/{runId}/points/values:
  *   get:
- *     operationId: List input points
- *     description: Return list of input and bidirectional points for a site, bidirectional values are excluded
+ *     summary: Get values for all points
  *     tags:
- *       - Site
+ *       - Run
  *     parameters:
- *       - $ref: '#/components/parameters/SiteID'
+ *       - $ref: '#/components/parameters/runId'
  *     responses:
  *       200:
- *         description: Points response
+ *         description: 'OK'
  *         content:
  *           application/json:
  *             schema:
- *               type: object
- *               properties:
- *                 data:
- *                   type: array
- *                   items:
- *                     $ref: '#/components/schemas/Point'
- *             example:
- *               data:
- *                 - id: c5bfacf7-bd44-4061-a44c-940bb71ba91f
- *                   name: CaseDefrostStatus
- *                   type: BIDIRECTIONAL
- *                 - id: a26d37f8-b479-46d2-8b0f-7e686be22223
- *                   name: MasterEnable
- *                   type: INPUT
+ *               type: array
+ *               items:
+ *                 $ref: '#/components/schemas/pointValue'
  */
-router.get("/sites/:siteId/points/inputs", (req, res) => {
-  const { siteId } = req.params;
-
-  const error = validate(
-    { siteId },
-    {
-      siteId: "required|uuid"
-    }
-  );
-  if (error) return res.status(400).json({ error });
-
+router.get("/runs/:runId/points/values", (req, res, next) => {
   api
-    .listPoints(siteId, /^(INPUT|BIDIRECTIONAL)$/)
-    .then((data) => res.json({ data }))
-    .catch(() => res.sendStatus(500));
+    .getPointsByRun(req.run)
```

Comment:
This should probably constrain to types output and bidirectional. Unless we want to allow users to query the current values of inputs. Which wouldn't be the worst thing.